src/renderer/views/wallet/SettingsView.tsx

```diff
@@ -337,13 +337,21 @@ export const SettingsView: React.FC = (): JSX.Element => {
     getLedgerErrorDescription(btcLedgerAddress, 'BTC')
   }, [bnbLedgerAddress, btcLedgerAddress, intl, removeLedgerAddress])
 
+  const setNetwork = useCallback(
+    (network: Network) => {
+      clearSavedBalances()
```

Comment:
Let's handle network on service layer. Whenever an user change the network, `network$` will trigger a change and we can reset balance state at service layer.

Reply:
> we will have circular dependency
If we have a circular dependency, let's re-organize modules

Reply:
> he other option is to set "infinite" subscription at the wallet/balances like

> ```
> export const clearSavedBalances = () => {
>   balanceRecord = {}
> }
> network$.subscribe(clearSavedBalances)
> ```

This looks good ^ btw.